APISyncExternalModule.php

```diff
@@ -918,6 +926,70 @@ function getRecordIdField($pid = null){
 
 		return $row['field_name'];
 	}
+	
+	private function translateFormNames(&$data) {
+		// get translations from module log table
+		$result = $this->queryLogs("SELECT translations WHERE message = ?", [
+			'form-translations'
+		]);
+		
+		// attempt to decode JSON to translations array
+		$translations = json_decode(db_fetch_assoc($result)['translations']);
+		if (!$translations) {
+			throw new \Exception("API Sync module failed to decode JSON to translations array before importing data");
+		}
+		
+		// translate [redcap_repeat_instrument] and [$form . '_complete'] fields where applicable
+		foreach ($data as &$instance) {
+			foreach ($translations as $local_name => $foreign_names) {
+				// [redcap_repeat_instrument] component
+				if (in_array($instance['redcap_repeat_instrument'], $foreign_names, true)) {
+					$instance['redcap_repeat_instrument'] = $local_name;
+				}
+				
+				// [$form . '_complete'] components
+				foreach($foreign_names as $foreign_form_name) {
+					if (isset($instance[$foreign_form_name . '_complete'])) {
+						$instance[$local_name . '_complete'] = $instance[$foreign_form_name . '_complete'];
+						unset($instance[$foreign_form_name . '_complete']);
+					}
+				}
```

Comment:
This seems like it's on the right track.  However, it is hard to assess without being tied to project configuration module settings, or the cron methods.  I still think creating a unit test for `prepareImportData()` and adding this code there (or in a method called from there) would be a really easy way to create a good test case.  Assuming that works out how I think it will, I still think we should rename `prepareImportData()` to `prepareDataForSend()` or something that is not import specific.